src/thorin/transform/resolve_loads.cpp

```diff
@@ -0,0 +1,189 @@
+#include "thorin/analyses/scope.h"
+#include "thorin/analyses/schedule.h"
+#include "thorin/world.h"
+
+namespace thorin {
+
+static void replace_ptr_uses(const Def* ptr) {
+    for (auto& use : ptr->uses()) {
+        if (auto store = use->isa<Store>()) {
+            store->replace(store->mem());
+        } else if (use->isa<Load>()) {
+            assert(false);
+        } else if (use->isa<LEA>()) {
+            replace_ptr_uses(use.def());
+        } else if (use->isa<Bitcast>()) {
+            replace_ptr_uses(use.def());
```

Comment:
You can also just write `replace_ptr_uses(use);`. `Use` has an appropriate cast available.